js/ui/tree_view/ui.tree_view.base.js

```diff
@@ -1158,36 +1158,46 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
             return true;
         }
 
-        if(!value && this._isLastRequired(node)) {
+        if(value === false && this._isLastRequired(node)) {
             if(this._showCheckboxes()) {
                 const $node = this._getNodeElement(node);
-                const checkbox = this._getCheckBoxInstance($node);
-
-                checkbox && checkbox.option('value', true);
+                this._setCheckboxValue($node, true);
             }
             return false;
         }
 
-        const selectedNodesKeys = this.getSelectedNodeKeys();
-        if(this._isSingleSelection() && value) {
-            each(selectedNodesKeys, (index, nodeKey) => {
-                this.unselectItem(nodeKey);
+        if(value === true && this._isSingleSelection()) {
+            const selectedKeys = this.getSelectedNodeKeys();
+            each(selectedKeys, (index, key) => {
+                this._dataAdapter.toggleSelection(key, false);
+                this._updateItemsUI();
+
+                this._fireItemSelectionChanged(this._getNode(key));
+                this._fireSelectionChanged();
             });
         }
 
         this._dataAdapter.toggleSelection(node.internalFields.key, value);
         this._updateItemsUI();
 
+        this._fireItemSelectionChanged(node, dxEvent);
+        this._fireSelectionChanged();
+        return true;
+    },
+
```

Comment:
Why the new '_setCheckboxValue' function is better than "this._getCheckBoxInstance($node).option('value', nodeSelection)"?

Reply:
Previously, I think, that it is easy to read. Complexity of this method was decreased.

But in th Treeview there is another method "changeCheckBoxState" with different behavior. And as result there are 2 methods with similar name and different behavior. So, summary complexity in the Treeview was increased.

So, I removed my new method.